e2e/tests/functional/plugins/conditionSet/conditionSet.e2e.spec.js

```diff
@@ -475,3 +475,110 @@ test.describe('Basic Condition Set Use', () => {
     });
   });
 });
+
+test.describe('Condition Set Composition', () => {
+  let conditionSet;
+  let exampleTelemetry;
+
+  test.beforeEach(async ({ page }) => {
+    await page.goto('./', { waitUntil: 'domcontentloaded' });
+
+    // Create Condition Set
+    conditionSet = await createDomainObjectWithDefaults(page, {
+      type: 'Condition Set',
+      name: 'Condition Set with Telemetry and Conditions'
+    });
+
+    // Create Telemetry Object
+    exampleTelemetry = await createExampleTelemetryObject(page);
+
+    // Make Link from Telemetry Object to Overlay Plot
+    await page.locator('button[title="More actions"]').click();
+
+    // Select 'Create Link' from dropdown
+    await page.getByRole('menuitem', { name: 'Create Link' }).click();
+
+    // Search and Select for Condition Set within Create Modal
+    await page.getByRole('dialog').getByRole('searchbox', { name: 'Search Input' }).click();
+    await page
+      .getByRole('dialog')
+      .getByRole('searchbox', { name: 'Search Input' })
+      .fill(conditionSet.name);
+    await page
+      .getByRole('treeitem', { name: new RegExp(conditionSet.name) })
+      .locator('a')
+      .click();
```

Comment:
I think there's an easier way to do this now? Check with Playwright's tooling. The generated locator should work and be simpler